cmd/operator/main.go

```diff
@@ -115,6 +133,8 @@ func init() {
 	flagset.StringVar(&cfg.TLSConfig.CertFile, "cert-file", "", " - NOT RECOMMENDED FOR PRODUCTION - Path to public TLS certificate file.")
 	flagset.StringVar(&cfg.TLSConfig.KeyFile, "key-file", "", "- NOT RECOMMENDED FOR PRODUCTION - Path to private TLS certificate file.")
 	flagset.StringVar(&cfg.TLSConfig.CAFile, "ca-file", "", "- NOT RECOMMENDED FOR PRODUCTION - Path to TLS CA file.")
+	flagset.StringVar(&cfg.HostTLSConfig.CertFile, "host-cert-file", "", "Path to certificate file to host web endpoint")
+	flagset.StringVar(&cfg.HostTLSConfig.KeyFile, "host-key-file", "", "Path to key file to host web endpoint")
```

Comment:
I wonder if we should even handle this here or just ask people to put the [kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy) in front of this. Getting into TLS is hairy business.

Reply:
I'm certain that I don't know all the cases you're thinking of, but my intention here was to provide the simplest way to do this set-up. I think that if someone wants to perform certificate rotation they can just restart the pod, since it doesn't have a requirement for being up 100% of the time.

A case I can think of that would become awkward is if the validation endpoint should be segregated from the rest of the Prometheus Operator API, with one requiring being served over TLS and the other not. 

There isn't anything preventing someone from using a proxy in front of prometheus operator to achieve this if they want to.

Could you suggest some other reasons for why this is a bad idea?

Reply:
This would need the ability to allow TLS version, cipher suites, key rotation/reload, and many other things that are easy to get wrong. As a matter of fact we have already done those mistakes in kube-rbac-proxy.

> A case I can think of that would become awkward is if the validation endpoint should be segregated from the rest of the Prometheus Operator API, with one requiring being served over TLS and the other not.

Agreed, I think it should all be one. I don't see how that wouldn't be the case here.

> I think that if someone wants to perform certificate rotation they can just restart the pod, since it doesn't have a requirement for being up 100% of the time.

I think this dramatically changes with this being a blocking call to CRUD PrometheusRule objects.